tools/ci_build/build.py

```diff
@@ -2074,6 +2074,17 @@ def run_onnxruntime_tests(args, source_dir, ctest_path, build_dir, configs):
             if is_windows():
                 cwd = os.path.join(cwd, config)
 
+            if args.enable_transformers_tool_test and not args.disable_contrib_ops and not args.use_rocm:
+                # PyTorch is required for transformers tests, and optional for some python tests.
+                # Install cpu only version of torch when cuda is not enabled in Linux.
+                extra = [] if args.use_cuda and is_linux() else ["--index-url", "https://download.pytorch.org/whl/cpu"]
```

Comment:
Can we always pass the `--index-url`? From the [Get Started](https://pytorch.org/get-started/locally/) page, it is used for Linux CUDA and there is a link for all three CUDA versions.

- CUDA 11.8: https://download.pytorch.org/whl/cu118
- CUDA 12.1: https://download.pytorch.org/whl/cu121
- CUDA 12.4: https://download.pytorch.org/whl/cu124

Reply:
--index-url will change if pytorch changes cuda version (like to 12.6). 
If we do not set index URL, it will always be the latest version in pypi.